tests/Microsoft.Bot.Builder.Azure.Tests/CosmosDbStorageTests.cs

```diff
@@ -166,6 +168,62 @@ public async Task WrittingNullStoreItems_Throws()
             }
         }
 
+        // NOTE: THESE TESTS REQUIRE THAT THE COSMOS DB EMULATOR IS INSTALLED AND STARTED !!!!!!!!!!!!!!!!!
+        // For issue https://github.com/Microsoft/botbuilder-dotnet/issues/871
+        // See the linked issue for details. This issue was happening when using the CosmosDB
+        // data store for state. The stepIndex, which was an object being cast to an Int64
+        // after deserialization, was throwing an exception for not being Int32 datatype.
+        // This test checks to make sure that this error is no longer thrown.
+        [TestMethod]
+        public async Task WaterfallCosmos()
+        {
+            if (CheckEmulator())
+            {
+                var convoState = new ConversationState(_storage);
+
+                var adapter = new TestAdapter()
+                    .Use(convoState);
+
+                var dialogState = convoState.CreateProperty<DialogState>("dialogState");
+                var dialogs = new DialogSet(dialogState);
+                dialogs.Add(new WaterfallDialog("test", new WaterfallStep[]
+                {
+                    async (dc, step, ct) =>
+                    {
+                        Assert.AreEqual(dc.ActiveDialog.State["stepIndex"].GetType(), typeof(Int32));
+                        await dc.Context.SendActivityAsync("step1"); return Dialog.EndOfTurn;
+                    },
+                    async (dc, step, ct) =>
```

Comment:
I suspect steps 2 and (for sure) step 3 are not needed in this test. Please confirm, and remove if they can be removed.